compiler/semstmts.nim

```diff
@@ -1153,8 +1153,8 @@ proc typeSectionFinalPass(c: PContext, n: PNode) =
       var x = a[2]
       while x.kind in {nkStmtList, nkStmtListExpr} and x.len > 0:
         x = x.lastSon
-      if x.kind notin {nkObjectTy, nkDistinctTy, nkEnumTy, nkEmpty} and
```

Comment:
This was carefully constructed to only cover the "nominal" types, ``ptr`` and ``ref`` are structural types.

Reply:
Okay there might be a better fix then, but currently ptr and ref are kind of evaluated twice - for eg. https://github.com/nim-lang/Nim/pull/10125#issuecomment-450472121